tensorboard/pip_package/pip_smoke_test.sh

```diff
@@ -0,0 +1,172 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+#
+# Smoke test for building, installing and basic usage of tensorboard pip package.
+#
+# Usage:
+#   pip_smoke_test.sh [--python3] [--retries <NUM_RETRIES>] [--port <PORT>]
+
+set -eu
+
+die() {
+  printf >&2 '%s ' "$@"
+  exit 1
+}
+
+PY_VERSION=2
+TEST_PORT=6006
+NUM_RETRIES=20
+while [[ "$#" -gt 0 ]]; do
+  if [[ "$1" == "--python3" ]]; then
+    PY_VERSION=3
+  elif [[ "$1" == "--retries" ]]; then
+    NUM_RETRIES="$2"
+    shift
+  elif [[ "$1" == "--port" ]]; then
+    TEST_PORT="$2"
+    shift
+  else
+    die "ERROR: Unrecognized argument $1"
+  fi
+  shift
+done
+
+echo
+echo "=== Performing smoke test of tensorboard PIP package ==="
+echo "Settings:"
+echo "  PY_VERSION=${PY_VERSION}"
+echo "  TEST_PORT=${TEST_PORT}"
+echo "  NUM_RETRIES=${NUM_RETRIES}"
+echo
+
+# Check that virtualenv is installed.
+if [[ -z "$(which virtualenv)" ]]; then
+  die "ERROR: virtualenv is required, but does not appear to be installed."
+fi
+
+PIP_TMP_DIR=$(mktemp -d --suffix _tensorboard)
+
+echo
+echo "Building tensorboard pip package in directory: ${PIP_TMP_DIR}"
+echo
+
+bazel build tensorboard/pip_package:build_pip_package
+
+# Create virtualenv directory, cleanly (i.e., no --system-site-packages).
+VENV_TMP_DIR=$(mktemp -d --suffix _tensorboard_venv)
+
+echo
+echo "Creating virtualenv directory at: ${VENV_TMP_DIR}"
+echo
+
+if [[ ${PY_VERSION} == 2 ]]; then
+  virtualenv -p python "${VENV_TMP_DIR}"
+  TF_NIGHTLY_URL='https://ci.tensorflow.org/view/Nightly/job/nightly-matrix-cpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=cpu-slave/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp27-none-linux_x86_64.whl'
+elif [[ ${PY_VERSION} == 3 ]]; then
+  virtualenv -p python3 "${VENV_TMP_DIR}"
+  TF_NIGHTLY_URL='https://ci.tensorflow.org/view/Nightly/job/nightly-matrix-cpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON3,label=cpu-slave/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp34-cp34m-linux_x86_64.whl'
+fi
+
+echo
+echo "Activating virtualenv at ${VENV_TMP_DIR}"
+echo
+
+export VIRTUAL_ENV="${VENV_TMP_DIR}"
+export PATH="${VENV_TMP_DIR}/bin:${PATH}"
+unset PYTHON_HOME
+
+echo
+echo "Performing pip required for building wheels"
+echo
+
+pip install upgrade pip setuptools wheel
+
+echo
+echo "Creating tensorboard pip package in directory: ${PIP_TMP_DIR}"
+echo
+
+bazel-bin/tensorboard/pip_package/build_pip_package "${PIP_TMP_DIR}"
```

Comment:
Hmm, so this has to be run in the root directory of the repository? It would be nice if that weren't the case, especially because this file isn't (and shouldn't be) at the root directory.

You can use `git rev-parse --show-toplevel` from anywhere in the repository, so adding `cd "$(git -C "$(dirname "$0")" rev-parse --show-toplevel)"` to the top of the script would suffice to fix this, I think.